code/modules/awaymissions/mission_code/TheFactory.dm

```diff
@@ -0,0 +1,902 @@
+		playsound(get_turf(src), chosen_sound, 50, TRUE, 0)
+		visible_message("<span class='danger'>[src] primes the grenade!.</span>")
+		addtimer(CALLBACK(src, .proc/prime), det_time)
+	else
+		return
+
+/mob/living/simple_animal/hostile/zombie_suicide/proc/prime()
+	explosion(src.loc,0,2,3,flame_range = 3)
+	new /obj/effect/gibspawner/generic(get_turf(src), src)
+	qdel(src)
+
+/mob/living/simple_animal/hostile/zombie_suicide/death(gibbed)
+	playsound(src, 'sound/creatures/szombiedeath.ogg', 60, TRUE)
+	..()
+
+/obj/item/grenade/syndieminibomb/concussion/frag/activated
+	det_time = 30
+
+/obj/item/grenade/syndieminibomb/concussion/frag/activated/Initialize()
+	..()
+	preprime()
+
+/mob/living/simple_animal/hostile/syndicate/factory/boss
+	name = "The Director"
+	desc = "This thing looks more like a machine than human."
+	maxHealth = 500
+	health = 500
+	icon_state = "facboss"
+	icon_living = "facboss"
+	see_in_dark = 13
+	vision_range = 12
+	aggro_vision_range = 12
+	search_objects = 1
+	minbodytemp = 0
+	speak = null
+	dodging = FALSE
+	mob_biotypes = list(MOB_ROBOTIC)
+	obj_damage = 100
+	move_force = MOVE_FORCE_EXTREMELY_STRONG
+	move_resist = MOVE_FORCE_EXTREMELY_STRONG
+	pull_force = MOVE_FORCE_EXTREMELY_STRONG
+	hardattacks = TRUE
+	melee_damage = 50
+	speed = 5
+	move_to_delay = 3
+	ranged = TRUE
+	hud_possible = list(ANTAG_HUD)
+	approaching_target = TRUE
+	ranged_ignores_vision = TRUE
+	environment_smash = ENVIRONMENT_SMASH_RWALLS
+	ranged_cooldown_time = 30
+	check_friendly_fire = 1
+	turns_per_move = 2
+	spacewalk = TRUE
+	rapid_melee = 0
+	lose_patience_timeout = 400
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	unsuitable_atmos_damage = 0
+	attack_sound = 'sound/weapons/sonic_jackhammer.ogg'
+	projectilesound = 'sound/weapons/shotgunshot.ogg'
+	var/cooldown = 0
+	loot = list(/obj/item/gun/ballistic/shotgun/lever_action,
+				/obj/effect/mob_spawn/human/corpse/facboss)
+
+/mob/living/simple_animal/hostile/syndicate/factory/boss/Shoot()
+	var/allowed_projectile_types = list(/obj/item/ammo_casing/shotgun/beanbag,
+										 /obj/item/ammo_casing/shotgun, /obj/item/ammo_casing/shotgun/incendiary,
+										 /obj/item/ammo_casing/shotgun/dragonsbreath,
+										 /obj/item/ammo_casing/shotgun/meteorslug,
+										 /obj/item/ammo_casing/shotgun/pulseslug,
+										 /obj/item/ammo_casing/shotgun/frag12,
+										 /obj/item/ammo_casing/shotgun/buckshot,
+										 /obj/item/ammo_casing/shotgun/rubbershot,
+										 /obj/item/ammo_casing/shotgun/incapacitate,
+										 /obj/item/ammo_casing/shotgun/improvised,
+										 /obj/item/ammo_casing/shotgun/ion,
+										 /obj/item/ammo_casing/shotgun/laserslug,
+										 /obj/item/ammo_casing/shotgun/breacher)
+	casingtype = pick(allowed_projectile_types)
+	..()
+	sleep(5)
+	playsound(get_turf(src), 'sound/weapons/shotgunpump.ogg', 50, 0, 0)
```

Comment:
You should probably add this on a timer instead of using sleep()

Reply:
I know, but when I try to make a timer for playsound it never works. And I don't think it's a good idea to make a whole new proc for one sound.